src/main/resources/com/google/api/codegen/py/main.snip

```diff
@@ -112,27 +113,44 @@
                 client_info = (
                     google.api_core.gapic_v1.client_info.DEFAULT_CLIENT_INFO)
             client_info.gapic_version = _GAPIC_LIBRARY_VERSION
+            self._client_info = client_info
 
             @# Parse out the default settings for retry and timeout for each RPC
             @# from the client configuration.
             @# (Ordinarily, these are the defaults specified in the `*_config.py`
             @# file next to this one.)
-            method_configs = google.api_core.gapic_v1.config.parse_method_configs(
+            self._method_configs = google.api_core.gapic_v1.config.parse_method_configs(
                 client_config['interfaces'][self._INTERFACE_NAME],
             )
 
-            @# Write the "inner API call" methods to the class.
-            @# These are wrapped versions of the gRPC stub methods, with retry and
-            @# timeout configuration applied, called by the public methods on
-            @# this class.
-            @join apiMethod : xapiClass.apiMethods
-                self._{@apiMethod.name} = google.api_core.gapic_v1.method.wrap_method(
-                    self.{@apiMethod.stubName}.{@apiMethod.grpcMethodName},
-                    default_retry=method_configs['{@apiMethod.grpcMethodName}'].retry,
-                    default_timeout=method_configs['{@apiMethod.grpcMethodName}'].timeout,
-                    client_info=client_info,
-                )
+            self._stubs = {}
+
+        def _intercept_channel(self, *interceptors):
+            """ Experimental. Bind gRPC interceptors to the gRPC channel.
+
+            Args:
+                interceptors (*Union[grpc.UnaryUnaryClientInterceptor, grpc.UnaryStreamingClientInterceptor, grpc.StreamingUnaryClientInterceptor, grpc.StreamingStreamingClientInterceptor]):
+                  Zero or more gRPC interceptors. Interceptors are given control in the order
+                  they are listed.
+            Raises:
+                TypeError: If interceptor does not derive from any of
+                  UnaryUnaryClientInterceptor,
+                  UnaryStreamClientInterceptor,
+                  StreamUnaryClientInterceptor, or
+                  StreamStreamClientInterceptor.
+            """
+            self.channel = grpc.intercept_channel(self.channel, *interceptors)
+            @join stub : xapiClass.stubs
+                self.{@stub.name} = (
+                    {@stub.grpcClientTypeName}(self.channel))
             @end
+            @if xapiClass.hasLongRunningOperations
+                @# Operations client for methods that return long-running operations
+                @# futures.
+                self.operations_client = (
```

Comment:
Can this be a private member?
`self._operations_client`

Reply:
Changed to be private.